core/src/test/scala/io/gearpump/util/GraphSpec.scala

```diff
@@ -178,6 +178,30 @@ class GraphSpec extends PropSpec with PropertyChecks with Matchers {
     assert(graph.hasCycle())
   }
 
+  property("topologicalOrderIterator " +
+    "and topologicalOrderWithCirclesIterator method should return equal order of graph with no circle") {
+    val graph = Graph.empty[String, String]
+    val defaultEdge = "edge"
+    graph.addVertex("A")
+    graph.addVertex("B")
+    graph.addVertex("C")
+    graph.addEdge("A", defaultEdge, "B")
+    graph.addEdge("B", defaultEdge, "C")
+
+    val topoNoCircles = graph.topologicalOrderIterator
+    val topoWithCircles = graph.topologicalOrderWithCirclesIterator
+
+    assert(topoNoCircles.zip(topoWithCircles).forall(x => x._1 == x._2))
+  }
+
+  property("Topological sort of graph with circles should work properly") {
+    val graph = Graph(1 ~> 3 ~> 4 ~> 6 ~> 5, 4 ~> 1, 1 ~> 2 ~> 4)
```

Comment:
Can you add a test case which have two strong connected components, and one component ranks over the other by topology order. 

So that we can examine the behavior for multiple cycles.